Vip Cars Rails Frontend

PraisesPJMT opened the pull request:
# Vip Cars React Frontend
## Project Description
Create a React app to serve as a client-side for the `Vip Cars Rails` project.
## Project Links
- `Vip Cars Rails Backend` PR Link: [Backend APP PR](https://github.com/edi-sipka/Vip_Cars_Rails/pull/33)
- `Vip Cars Rails Backend` APP Link: [Backend App](https://github.com/edi-sipka/Vip_Cars_Rails)
## Feature Description
- Created a user `Registration` Page
- Created a user `Log In` Page
- Created a `Home` Page to display all VIP cars
- Created a `Details` Page to display  details of a VIP cars
- Created a `Add Car` Page for `admin`s to add a VIP cars
- Created a `Delete Car` Page for `admin`s to delete a VIP cars
- Created a `ReservePage` Page for users to reserve a VIP cars
- Created a `Reservations` Page for users to see all VIP cars reservations
## Team Members
The following team members worked on the `Vip Cars Rails` project:
- Edi Sipka (@edi-sipka)
- Praises Tula (@PraisesPJMT)
- Youta Christabelle (@Christabelle143)
- Okolike Assumpta (@summyalena)
- Nonhlanhla (@29td)



Happy Reviewing!!!!!

29td committed:
Added navigation

29td committed:
Added footer

29td committed:
Installed bootstrap

PraisesPJMT committed:
Merge pull request #3 from edi-sipka/navbar

Navbar

Christabelle143 committed:
install react-router-dom

Christabelle143 committed:
install react-redux

PraisesPJMT committed:
Add `Redux` dependencies

PraisesPJMT committed:
Add `User` api functions

PraisesPJMT committed:
Add `User` store

PraisesPJMT committed:
Add `User` API functions

PraisesPJMT committed:
Add `Car` store

PraisesPJMT committed:
Fix `User` api bug

PraisesPJMT committed:
Add `Reservations` api functions

PraisesPJMT committed:
Add `Reservations` reducer

PraisesPJMT committed:
Add `Reservations` store

PraisesPJMT committed:
Merge pull request #4 from edi-sipka/redux-store

Add Redux Store

Christabelle143 committed:
install reduxjs/toolkit

Christabelle143 committed:
create basic components

Christabelle143 committed:
add the configureStore.js file

Christabelle143 committed:
fix linters errors

Christabelle143 committed:
Merge branch 'dev' into basic-components

Christabelle143 committed:
fix linters errors

PraisesPJMT committed:
Merge pull request #5 from edi-sipka/basic-components

Basic components

PraisesPJMT committed:
Add react-router-dom dependencies

PraisesPJMT committed:
Add routes to app

PraisesPJMT committed:
Update `User` API function

PraisesPJMT committed:
Update `User` reducer

PraisesPJMT committed:
Add User `RegistrationPage` component

PraisesPJMT committed:
Add User `LoginPage` component

PraisesPJMT committed:
Improve user experience

PraisesPJMT committed:
Fix linter errors

PraisesPJMT committed:
Merge `dev` branch

PraisesPJMT committed:
Fix logout bug

PraisesPJMT committed:
Merge pull request #6 from edi-sipka/user-log

Add User Registration & Signing

29td committed:
Added main page

29td committed:
Added mainpage

PraisesPJMT committed:
Update api functions

PraisesPJMT committed:
Update `Car` reducer

PraisesPJMT committed:
Update `User` reducer

PraisesPJMT committed:
Add `AddCar` component

PraisesPJMT committed:
Improve user experience

PraisesPJMT committed:
Update api functions

PraisesPJMT committed:
Merge pull request #7 from edi-sipka/add-car

Add Car Addition Feature

29td committed:
Created mainPg

PraisesPJMT committed:
Add `Swiper` dependencies

PraisesPJMT committed:
Add api and action to delete car

PraisesPJMT committed:
Add `DeleteCar` component to app

PraisesPJMT committed:
Update `DeleteCar` component to app

29td committed:
Added Home page

29td committed:
Added reservation

29td committed:
Added reservation

29td committed:
Changed the API_URL source

29td committed:
Fix linter errors

edi-sipka committed:
Merge pull request #9 from edi-sipka/main-page

Main page

PraisesPJMT committed:
Merge branch 'dev' into remove-car

PraisesPJMT committed:
Fix app bugs

PraisesPJMT committed:
Merge pull request #8 from edi-sipka/remove-car

Add Car Deletion Feature

Christabelle143 committed:
start the detail page

Christabelle143 committed:
add details for each car

Christabelle143 committed:
add button for car reservation

Christabelle143 committed:
fix linters errors

Christabelle143 committed:
add the pounds sign for he currency

Christabelle143 committed:
style the title for the detail page

Christabelle143 committed:
align the details-infos

Christabelle143 committed:
display the details-infos for a particualr car as a table

Christabelle143 committed:
add background color grey on the details infos following the design

Christabelle143 committed:
add styling to the back button

PraisesPJMT committed:
Fix page loading

Christabelle143 committed:
fix linters errors

PraisesPJMT committed:
Merge pull request #10 from edi-sipka/detailpage

Detailpage

PraisesPJMT committed:
Merge pull request #11 from edi-sipka/loading

Fix Page Loading

29td committed:
Working on a mobile version

29td committed:
Added MobileMenu functionality

PraisesPJMT committed:
Add unit tests and update `README.md` file

PraisesPJMT committed:
Merge pull request #12 from edi-sipka/testing

Add Unit Tests and Update `README.md` file

Meltrust commented on README.md:
```diff
@@ -1,154 +1,223 @@
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
+This project is [MIT](./MIT.md) licensed.
```
- [ ]  On the "License" section of your readme, the link is broken because it points to a file that does not exist in your project.

## Tips and extra help

Go [here](https://github.com/microverseinc/readme-template), grab the `MIT.md` file from that repo, paste it on the root of your project, and link to it in the "License" section of your readme.

That way, your app will be well documented 💪

PraisesPJMT replied:
Well noted! The Licence link is now fixed!

Meltrust commented on src/components/Detail.css:
```diff
@@ -0,0 +1,62 @@
+.details-container {
```
- [ ] On the mobile view, I couldn't find a way to close the navbar:

![image](https://user-images.githubusercontent.com/57421823/214326533-a2a8f294-96a1-44ce-9677-a4ca650bdb5a.png)


If it is indeed possible to close the navbar, please try to make it more obvious so your users can do it easily.  If this is by design, you may choose to leave it, however, this is **strongly not recommended** because the area for the content is vastly reduced, making the UX suffer a little. 👍

PraisesPJMT replied:
Well noted! The NavBar bug for the mobile menu is now fixed!
Thanks so much for the Observation and Recommendation!!

Meltrust reviewed (changes requested):
Hi @PraisesPJMT,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

29td committed:
Added styling for the Mobile

29td committed:
Fix linters

PraisesPJMT committed:
Update `README.md` file

PraisesPJMT committed:
Merge branch 'dev' into mobile

PraisesPJMT committed:
Fix linter error

PraisesPJMT committed:
Merge pull request #14 from edi-sipka/mobile

Mobile

PraisesPJMT commented:
## Code Review
Thanks so much @Meltrust for the amazing review and observation. 
## Effected Changes
Based on the previous code review by @Meltrust, the following changes were made in line with recommendation;
- Fixed link to the MIT Licence of the `README.md` file
- Fixed menu closing for mobile view


Happy Reviewing!!!!!!!!!!!!!!

PraisesPJMT committed:
Add remote API URL

PraisesPJMT committed:
Add `_render` file to handle netlify routes

Meltrust reviewed (approved):
_note:  This review outcome applies only to the front-end project.  In order to see the backend review comments, kindly go to that [PR](https://github.com/edi-sipka/Vip_Cars_Rails/pull/33)._

#### Hi @PraisesPJMT,


# **Wow, you did it 🎉**
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
![applause](https://media.giphy.com/media/srg19CG0cKMuI/giphy.gif)

### To highlight

- The navbar bug has been fixed✔️ 
- License has been implemented and the link has been fixed✔️ 
- Beautiful frontend✔️
- Splendid job ✔️  



Cheers and Happy coding!👏👏👏


------

- [x] OPTIONAL: This is just a heads-up for you.  When I came back to this project to re-review it, I was greeted with a white screen.  

I tried to investigate the reason:

- Upon opening the console, I found the error `cars.map is not a function`.  (It seems 'cars' was undefined at that moment.)
- After that, I opened the "Network" tab, and noticed a "401 unauthorized" in "GET cars".  (Rails refused to fetch the cars).
- Upon that, I reached to the "response" tab and got "Token is expired" from the server. (This was the reason for the whole white-screen drama.  My token was no longer valid.).
- In order to make the frontend work again so I could review, I had to go into dev-tools and delete my local storage:

![image](https://user-images.githubusercontent.com/57421823/214560912-4db96dd7-d58c-453b-ad0d-ea599ef65374.png)

- After doing this, the app worked again because the token was created again fresh.

Tagging this as "OPTIONAL", since it was not included in the last review (my token was brand new in the first review).

I'm leaving this comment as a heads-up to you so that if this ever happens to you, you can take a look at it and know the reason.

PraisesPJMT committed:
Add deployment `build` files

PraisesPJMT committed:
Modify doplyment settings

PraisesPJMT committed:
Modify build settings in package.json

PraisesPJMT committed:
Update README.md file

PraisesPJMT committed:
Fix `User` session expire bug

PraisesPJMT committed:
Merge pull request #15 from edi-sipka/deployment

Deployment

PraisesPJMT committed:
Change API to local server URL

PraisesPJMT commented:
## Code Review
Thanks so much @Meltrust for the amazing review and observation. 
## Effected Changes
Based on the previous code review by @Meltrust, the following changes were made in line with recommendation;
- Fixed expired `User` session bug


Happy Reviewing!!!!!!!!!!!!!!
